### PR TITLE
Introduce MessagePort on AudioWorkletNode/Processor

### LINKED ISSUE #1194: [audioworklet] AudioWorkletProcessor ondata should just be a callback rather than an EventHandler
I believe the intention here was to specifically deliver any ondata's sent from the main thread directly before the process method is called for that node? 

If so AudioWorkletProcessor shouldn't have `ondata` being a `EventHandler` just a callback similar to `process`.

### PR DESCRIPTION
**Pull request by hoch**
Preview: https://cdn.rawgit.com/hoch/web-audio-api/audioworklet-message-port/index.html

This fixes #1194.

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -6066,35 +6066,18 @@ <h2 id="AudioWorkletNode">
...
             </dt>
             <dd>
               <p>
-                The <code>ondata</code> handler is called whenever the
-                associated <a>AudioWorkletProcessor</a> sends data back to the
-                <a>AudioWorkletNode</a> instance.
+                Every <a>AudioWorkletNode</a> has an associated
+                <code>port</code> which is a <a href=
+                "https://html.spec.whatwg.org/multipage/comms.html#message-ports">
+                MessagePort</a>. It is connected to the other message port
```
Comment (rtoy) [inline index.html]: Sorry, I can't parse this sentence. Same issue below with the same description.

**Comment by hoch** [comment]
Yes, this fixes #1194. (as shown in the description)

File: index.html (hunk 1 above)
Comment (hoch) [inline index.html]: Okay, can you suggest something? We have two ports, one in AWN and the other in AWP. They are connected to each other.

**Comment by hoch** [comment]
@rtoy Thanks!
@joeberkovitz Could you take a look at this change?

File: index.html [hunk 2]
```diff
@@ -6066,35 +6066,17 @@ <h2 id="AudioWorkletNode">
...
             <dd>
               <p>
-                The <code>ondata</code> handler is called whenever the
-                associated <a>AudioWorkletProcessor</a> sends data back to the
-                <a>AudioWorkletNode</a> instance.
+                Every <a>AudioWorkletNode</a> has an associated
+                <code>port</code> which is a <a href=
+                "https://html.spec.whatwg.org/multipage/comms.html#message-ports">
+                MessagePort</a>. It is connected to the port on the
+                corresponding <a>AudioWorkletProcessor</a> object.
```
Comment (joeberkovitz) [inline index.html]: Suggest a little additional text here and on the other `MessagePort` stating that this is the way that Web Audio allows the AWN and AWP to send data to each other. Technically this is redundant but we could go a little further than just saying that the ports are connected.

**Comment by padenot** [comment]
LGTM, merging.

**Pull request merged**